app/lib/frontend/handlers.dart

```diff
@@ -28,21 +28,28 @@ void _logPubHeaders(shelf.Request request) {
 ///
 /// The passed in [shelfPubApi] handler will be used for handling requests to
 ///   - /api/*
-Future<shelf.Response> appHandler(
-  shelf.Request request,
-  shelf.Handler shelfPubApi,
-) async {
-  _logPubHeaders(request);
-
-  final redirected = tryHandleRedirects(request);
-  if (redirected != null) return redirected;
-
+shelf.Handler createAppHandler(shelf.Handler shelfPubApi) {
+  final pubDartlangOrgHandler = PubDartlangOrgService().router.handler;
   final pubSiteHandler = PubSiteService(shelfPubApi).router.handler;
+  return (shelf.Request request) async {
+    _logPubHeaders(request);
+
+    // do pub.dartlang.org-only routes
+    if (request.requestedUri.host == legacyHost) {
+      final rs = await pubDartlangOrgHandler(request);
+      if (rs != null) {
+        return rs;
+      }
+    }
+
+    final redirected = tryHandleRedirects(request);
```

Comment:
Why don't we move `PubDartlangOrgService` into `redirects.dart` and handle those redirects in `tryHandleRedirects`?

Reply:
We can rearrange that after the other domain-specific redirects are migrated, but most of all I really like the fact that all routes are in a single file. I think it is better than having two or more files based on the domain name...